packages/modular-scripts/react-scripts/config/webpack.config.js

```diff
@@ -735,20 +742,28 @@ module.exports = function (webpackEnv) {
   return webpackConfig;
 };
 
-function createExternalDependenciesMap(externalDependencies) {
+function createExternalDependenciesMap(
+  externalDependencies,
+  externalResolutions,
+) {
   const externalCdnTemplate =
     process.env.EXTERNAL_CDN_TEMPLATE ||
     'https://cdn.skypack.dev/[name]@[version]';
 
-  return Object.entries(externalDependencies).reduce(
-    (acc, [name, version]) => ({
+  return Object.entries(externalDependencies).reduce((acc, [name, version]) => {
+    if (!externalResolutions[name]) {
+      throw new Error(
+        `Dependency ${name} found in package.json but not in lockfile. Have you installed your dependencies?`,
+      );
+    }
+    return {
       ...acc,
       [name]: externalCdnTemplate
         .replace('[name]', name)
-        .replace('[version]', version),
-    }),
-    {},
-  );
+        .replace('[version]', version || externalResolutions[name])
```

Comment:
Is there a case where version would not be defined?

Reply:
> Is there a case where version would not be defined?

Good question. There is at the moment, because we relaxed our criteria to only warn in case a version is nor present in (on e of the) package.json. The reason is that some widely used libraries rely on transitive dependencies (dm me for more info)